src/FSharp.Analyzers/DisposedBeforeAsyncRunAnalyzer.fs

```diff
@@ -103,6 +103,9 @@ let collectUses (sourceText : ISourceText) (ast : ParsedInput) (checkFileResults
             || elseExpr |> Option.map hasAsyncOrTaskInBody |> Option.defaultValue false
         | SynExpr.TryFinally (tryExpr = tryExpr) -> hasAsyncOrTaskInBody tryExpr
         | SynExpr.TryWith (tryExpr = tryExpr) -> hasAsyncOrTaskInBody tryExpr
+        | SynExpr.Match (clauses = clauses) ->
```

Comment:
Maybe make a note that this whole thing isn't tail-recursive? Or that is at least my educated guess.

Should we include `SynExpr.MatchBang` here?

Reply:
Yeah, but in theory we could have something like:

```fsharp
let meh = async {
  let blah = new DisposableThing ()
  use blah = blah
  return! async {
      match! async {return 1} with
      | _ -> async { return "hi" }
  }
}
```

Might not be relevant for now.